crates/daemon/src/monitor.rs

```diff
@@ -1023,3 +1030,139 @@ static TRANSACTION_BROADCAST_COUNTER: conquer_once::Lazy<prometheus::IntCounterV
+        let client = client.clone();
+
+        tokio::task::spawn_blocking({
+            move || {
+                for script in batch {
+                    match client.script_get_history(&script) {
+                        Ok(script_history_response) => {
+                            // We use blocking_send to stay within a sync context here
+                            // One should not use async code in a spawn_blocking block
+                            if let Err(e) = tx_script_updates.blocking_send(script_history_response)
+                            {
+                                tracing::error!(
+                                    "Error when processing script_get_history response: {e:#}"
+                                )
+                            }
+                        }
+                        Err(e) => {
+                            tracing::error!("Error when fetching script history: {e:#}")
+                        }
+                    }
+                }
+            }
+        });
+    });
+
+    rx_script_updates
+}
+
+#[cfg(test)]
+mod test {
+    use crate::monitor::batch_script_get_history;
+    use bdk::bitcoin;
+    use bdk::bitcoin::Script;
+    use bdk::electrum_client;
+    use bdk::electrum_client::GetHistoryRes;
+    use std::str::FromStr;
+    use std::sync::Arc;
+    use std::time::Duration;
+    use std::time::SystemTime;
+    use tracing_subscriber::util::SubscriberInitExt;
+
+    fn get_test_server() -> String {
+        std::env::var("TEST_ELECTRUM_SERVER")
+            .unwrap_or_else(|_| "electrum.blockstream.info:50001".into())
+    }
+
+    /// Test sanity of batch_script_get_history by simulating what the production code does
+    ///
+    /// Ignored on CI because it requires a mainnet Electrum instance to run properly
+    #[tokio::test(flavor = "multi_thread", worker_threads = 10)]
+    #[ignore]
+    async fn given_many_scripts_then_batch_script_get_history_works() {
+        // Mt.Gox hack address
+        let script = bitcoin::Address::from_str("1FeexV6bAHb8ybZjqQMjJrcCrHGW9sb6uF")
+            .unwrap()
+            .script_pubkey();
+        let response_length_to_assert = 328;
+
+        let mut scripts = Vec::new();
+
+        for _ in 1..100 {
+            scripts.push(script.clone());
+        }
+
+        test_runner(scripts, response_length_to_assert).await;
+    }
+
+    #[tokio::test(flavor = "multi_thread", worker_threads = 10)]
+    #[ignore]
+    async fn given_script_not_on_chain_then_batch_script_get_history_returns_empty_response() {
+        // Some testnet address (to simulate a script where we get an empty response)
+        let script = bitcoin::Address::from_str("2MsQEtPJ6JJZszMYrD6udjUyTDFLczWQrv9")
+            .unwrap()
+            .script_pubkey();
+        let response_length_to_assert = 0;
+
+        test_runner(vec![script], response_length_to_assert).await;
+    }
+
+    async fn test_runner(scripts: Vec<Script>, assert_response_len: usize) {
```

Comment:
🔧 Should probably be called `test_batch_script_get_history` or something like that.